lib/std/fs.zig

```diff
@@ -864,6 +864,7 @@ pub const Dir = struct {
             .OBJECT_NAME_INVALID => unreachable,
             .OBJECT_NAME_NOT_FOUND => return error.FileNotFound,
             .OBJECT_PATH_NOT_FOUND => return error.FileNotFound,
+            .NO_MEDIA_IN_DEVICE => return error.FileNotFound,
```

Comment:
The error set for this function is `std.os.OpenError` which already includes `NoDevice` (from POSIX `ENODEV`). I think that would be perfect here! (I'll just do the fixup myself real quick)